src/libPMacc/include/cuSTL/cursor/accessor/LinearInterpAccessor.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2015 Heiko Burau
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "algorithms/math/defines/modf.hpp"
+
+namespace PMacc
+{
+namespace cursor
+{
+
+/** Performs a 1D, linear interpolation on access.
+ *
+ * \tparam T_Cursor 1D input data
+ * \tparam T_Position type of the weighting factor
+ */
+template<typename T_Cursor, typename T_Position = float>
+struct LinearInterpAccessor
```

Comment:
currently this functor is limited to 1d data. Please add an assert, implement N-dimensional interpolation or call is something like LinearInterpAccessor1D.